pins/boards.py

```diff
@@ -248,8 +250,16 @@ def _pin_store(
 
         pin_name = self.path_to_pin(name)
 
+        # Pre-emptively fetch the most recent pin's meta if it exists - this is used
+        # for the force_identical_write check
+        abort_if_identical = not force_identical_write and self.pin_exists(name)
```

Comment:
this *may* make writing pins slightly slower by making a call to the pin in `self.pin_exists()`. I think it's right move here (and I don't see a significant slowdown in tests at all), but just noting it if other calls get added and things start to feel sluggish